suitcase/json_metadata/__init__.py

```diff
@@ -8,30 +10,192 @@
+        kwargs to be passed to ``json.dump``.
+    Returns
+    -------
+    dest : dict
+        dict mapping the 'labels' to lists of file names
+    Examples
+    --------
+    Generate files with unique-identifer names in the current directory.
+    >>> export(gen, '')
+    Generate files with more readable metadata in the file names.
+    >>> export(gen, '', '{plan_name}-{motors}-')
+    Include the experiment's start time formatted as YY-MM-DD_HH-MM.
+    >>> export(gen, '', '{time:%%Y-%%m-%%d_%%H:%%M}')
+    Place the files in a different directory, such as on a mounted USB stick.
+    >>> export(gen, '/path/to/my_usb_stick')
+    """
     serializer = Serializer(directory, file_prefix, **kwargs)
     try:
         for item in gen:
             serializer(*item)
     finally:
         serializer.close()
 
-     return serializer.artifacts
+    return serializer.artifacts
 
 
 class Serializer(event_model.DocumentRouter):
-    def __init__(self, directory, file_prefix, **kwargs):
+    """
+    Serialize the metadata from a stream of documents to a JSON file.
+
+    This creates one file named ``<directory>/<file_prefix>meta.json``
+
+    The structure of the JSON is::
+    {'metadata': {'start': start_doc, 'stop': stop_doc,
+                  'descriptors': {stream_name1: {
+                                      'descriptor uid':descriptor_doc, ...},
+                                  stream_name2: ...}}}
+    .. note::
+
+        This can alternatively be used to write data to generic buffers rather
+        than creating files on disk. See the documentation for the
+        ``directory`` parameter below.
+
+    Parameters
+    ----------
+    directory : string, Path or Manager.
+        For basic uses, this should be the path to the output directory given
+        as a string or Path object. Use an empty string ``''`` to place files
+        in the current working directory.
+        In advanced applications, this may direct the serialized output to a
+        memory buffer, network socket, or other writable buffer. It should be
+        an instance of ``suitcase.utils.MemoryBufferManager`` and
+        ``suitcase.utils.MultiFileManager`` or any object implementing that
+        inferface. See the suitcase documentation (LINK ONCE WRITTEN) for
+        details.
+    file_prefix : str, optional
+        The first part of the filename of the generated output files. This
+        string may include templates as in ``{proposal_id}-{sample_name}-``,
+        which are populated from the RunStart document. The default value is
+        ``{uid}-`` which is guaranteed to be present and unique. A more
+        descriptive value depends on the application and is therefore left to
+        the user.
+    **kwargs : kwargs
+        kwargs to be passed to ``json.dump``.
+    Returns
+    -------
+    dest : dict
+        dict mapping the 'labels' to lists of file names
+    """
+    def __init__(self, directory, file_prefix='{uid}-', **kwargs):
```

Comment:
Headers may have numpy arrays in them. (It's relatively uncommon but does occur on the floor in places.) To handle that, you can [reuse this strategy](https://github.com/NSLS-II/suitcase-jsonl/pull/2/files). Thought instead of copy/pasting `NumpyEncoder` as I did it would be better to require `event-model >  v1.8.0rc2` (not rc1) and import `NumpyEncoder` from there.

Reply:
done